seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/SeataSagaAutoConfiguration.java

```diff
@@ -0,0 +1,121 @@
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import javax.sql.DataSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.concurrent.ThreadPoolExecutorFactoryBean;
+
+/**
+ * Saga auto configuration.
+ *
+ * @author wang.liang
+ */
+@Configuration
+@ConditionalOnProperty(StarterConstants.SEATA_PREFIX + ".saga.enabled")
+@AutoConfigureAfter(DataSourceAutoConfiguration.class)
+public class SeataSagaAutoConfiguration {
+
+    /**
+     * Create state machine config bean.
+     */
+    @Bean
+    @ConditionalOnBean(DataSource.class)
+    @ConditionalOnMissingBean
+    @ConfigurationProperties(StarterConstants.SEATA_PREFIX + ".saga.state-machine")
+    public StateMachineConfig dbStateMachineConfig(DataSource dataSource,
+        @Autowired(required = false) ThreadPoolExecutor threadPoolExecutor,
+        @Value("${spring.application.name:}") String applicationId,
+        @Value("${seata.tx-service-group:}") String txServiceGroup) {
+        DbStateMachineConfig config = new DbStateMachineConfig();
+        config.setDataSource(dataSource);
+        config.setApplicationId(applicationId);
+        config.setTxServiceGroup(txServiceGroup);
+
+        if (threadPoolExecutor != null) {
+            config.setThreadPoolExecutor(threadPoolExecutor);
+        }
+
+        return config;
+    }
+
+    /**
+     * Create state machine engine bean.
+     */
+    @Bean
+    @ConditionalOnMissingBean
+    public StateMachineEngine stateMachineEngine(StateMachineConfig config) {
+        ProcessCtrlStateMachineEngine engine = new ProcessCtrlStateMachineEngine();
+        engine.setStateMachineConfig(config);
+        new StateMachineEngineHolder().setStateMachineEngine(engine);
+        return engine;
+    }
+
+    @Configuration
+    @ConditionalOnProperty(name = StarterConstants.SEATA_PREFIX + ".saga.state-machine.enable-async", havingValue = "true")
+    @EnableConfigurationProperties({SeataSagaAsyncThreadPoolProperties.class})
+    static class SagaAsyncThreadPoolExecutorConfiguration {
```

Comment:
add rejectedExecutionHandler default CallerRunsPolicy

Reply:
I added it.